lib/public/views/Runs/format/formatRunDetectors.js

```diff
@@ -29,85 +29,81 @@ export const formatRunDetectors = (run, detailsModel) => {
-                    icon = iconBan();
-                    break;
-            }
-            return h(
-                `span.detector-badge.flex-row.justify-center.lh1.g1.br2.pv1.ph2.b1.b-${color}.${color}${isPresent ? '.isPresent' : ''}`,
-                { style: 'flex-basis: 5rem;' },
-                [h('.detector-name', name), h('.detector-quality-icon', icon)],
+        Success: (allDetectors) => {
+            const detectorsBadges = h(
+                '.g2.flex-row.flex-grow.flex-wrap.justify-end',
+                allDetectors.map(({ id, name }) => {
+                    const quality = detectorsQualities.find((x) => x.id == id) ? detectorsQualities.find((x) => x.id == id).quality : null;
+                    const actualQuality = detailsModel.isEditModeEnabled
+                        ? detailsModel.runPatch.getDetectorQuality(id) || quality
+                        : quality;
+                    let color;
+                    let icon;
+                    switch (actualQuality) {
+                        case RunDetectorQualities.GOOD:
+                            color = 'success';
+                            icon = iconCheck();
+                            break;
+                        case RunDetectorQualities.BAD:
+                            color = 'danger';
+                            icon = iconX();
+                            break;
+                        default:
+                            color = 'gray';
+                            icon = iconBan();
+                            break;
+                    }
+                    return h(
+                        `span.detector-badge.flex-row.justify-center.lh1.g1.br2.pv1.ph2.b1.b-${color}.${color}`,
+                        { style: 'flex-basis: 5rem;' },
+                        [h('.detector-name', name), h('.detector-quality-icon', icon)],
+                    );
+                }),
             );
-        }),
-    );
 
-    if (detailsModel.isEditModeEnabled && (run.timeTrgEnd || run.timeO2End)) {
-        detectorsBadges.children = detectorsBadges.children.filter((elem) => elem.attrs.className.includes('isPresent'));
-        return h(
-            '',
-            dropdown(
-                detailsModel.editionDetectorsDropdownModel,
-                detectorsBadges,
-                h(
-                    '.scroll-y',
-                    detectorsQualities.map(({ id, name, quality }) => h(
-                        '.flex-row.justify-between.g4.p2',
-                        [
-                            name,
-                            h(
-                                '.flex-row.g2.items-center',
-                                RUN_DETECTOR_QUALITIES.map((qualityOption) => h('label.form-check-label.flex-row.g2', [
+            if (detailsModel.isEditModeEnabled && (run.timeTrgEnd || run.timeO2End)) {
+                detectorsBadges.children = detectorsBadges.children.filter((elem) => !elem.attrs.className.includes('b-gray'));
```

Comment:
This is a nice workaround, but still: here you have a component directive that depend on a css class. This should never happen: component description should result in dom description, never the other way around.
Here you need to think from a higher point of view, and maybe split the component with the `isEditModeEnabled` earlier in two separated component
We can discuss about it if you want